histomicstk/features/compute_global_graph_features.py

```diff
@@ -0,0 +1,168 @@
+    poly_props = PolyProps._make(map(pop_stats, (areas, peris, max_dists)))
+
+    # Assume that each Voronoi vertex is on exactly three ridges.
+    ridge_points = vor.ridge_points
+    # This isn't exactly the collection of sides, since if they should
+    # be counted per-triangle then we weight border ridges wrong
+    # relative to ridges that are part of two triangles.
+    ridge_lengths = dist(*np.swapaxes(centroids[ridge_points], 0, 1))
+    sides = ridge_lengths
+    # Point indices of each triangle
+    tris = [[] for _ in vertices]
+    for p, ri in enumerate(vor.point_region):
+        for vi in vor.regions[ri]:
+            if vi != -1:
+                tris[vi].append(p)
+    # This will only fail in particular symmetrical cases where a
+    # Voronoi vertex is associated with more than three centroids.
+    # Since this should be unlikely in practice, we don't handle it
+    # beyond throwing an AssertionError
+    assert all(len(t) == 3 for t in tris)
+    tris = np.asarray(tris)
+    areas = np.stack(poly_area(centroids[t]) for t in tris)
+    tri_props = TriProps._make(map(pop_stats, (sides, areas)))
+
+    graph = sparse.coo_matrix((ridge_lengths, np.sort(ridge_points).T),
+                              (len(centroids), len(centroids)))
+    mst = minimum_spanning_tree(graph)
+    # Without looking into exactly how minimum_spanning_tree
+    # constructs its output, elimate any explicit zeros to be on the
+    # safe side.
+    mst_branches = pop_stats(mst.data[mst.data != 0])
+
+    tree = KDTree(centroids)
+    neigbors_in_distance = {
+        # Yes, we just throw away the actual points
+        r * scale:
+        pop_stats(np.stack(map(len, tree.query_ball_tree(tree, r * scale))) - 1)
+        for r in range(1, 6)
+    }
+    distance_for_neighbors = dict(zip(
+        neighbor_counts,
+        map(pop_stats, tree.query(centroids, [c + 1 for c in neighbor_counts])[0].T),
+    ))
+    density_props = DensityProps(neigbors_in_distance, distance_for_neighbors)
+
+    return Props(poly_props, tri_props, mst_branches, density_props)
+
+
+def poly_area(vertices):
+    return abs(poly_signed_area(vertices))
+
+
+def poly_signed_area(vertices):
+    return .5 * linalg.det(
+        np.stack((vertices, np.roll(vertices, -1, axis=-2)), -1)
+    ).sum(-1)
+
+
+def poly_peri(vertices):
+    return dist(vertices, np.roll(vertices, -1, axis=-2)).sum(-1)
+
+
+def dist(x, y):
+    """Compute the distance between two sets of points.  Has signature
+    (i),(i)->().
+
+    """
+    return (np.subtract(x, y) ** 2).sum(-1) ** .5
+
+
+def pop_stats(pop):
+    # Filter out outliers (here defined as points more than three
+    # standard deviations away from the mean)
+    while True:
+        mean = pop.mean()
+        stddev = pop.std()
+        mask = abs(pop - mean) <= 3 * stddev
+        if mask.all():
+            break
+        pop = pop[mask]
```

Comment:
`mean` and `stddev` are sensitive to outliers. 

Use [this](https://www.datacrucis.com/research/find-outliers-in-an-array.html) IQR based approach.

Also compute all population stats after outlier rejection.